handlers/protected-files/main.go

```diff
@@ -12,12 +12,28 @@ import (
 )
 
 const (
-	baseBranch = "master"
+	baseBranch             = "master"
+	PRCheckWorkflow        = "pr_check"
+	changesApprovedEnvName = "STEPLIB_CHANGES_APPROVED"
 )
 
 // Config ...
 type Config struct {
-	BypassGate bool `env:"BYPASS_GATE"`
+	IsApproved bool `env:"changes_approved"`
+}
+
+// BuildTriggerRequestModel ...
+type BuildTriggerRequestModel struct {
+	HookInfo    HookInfoModel    `json:"hook_info"`
+	BuildParams BuildParamsModel `json:"build_params"`
+	UserAgent   string           `json:"triggered_by,omitempty"`
```

Comment:
Although `UserAgent` is completely descriptive, we might consider using the same scheme for mapping properties to json fields.

If json field name is `triggered_by`, this property could be `TriggeredBy` accordingly.